### PR TITLE
Broken links fix - 1

### PR DESCRIPTION
**Pull request by nilminiwso2**
- Introducing base_path for URL specification
- Escaping `{{` instances because the base_path plugin uses `{{` 
- Fix broken fragments, links, and images

**Tested the following**
Tested the fragments, hyperlinks, and images in the following sections:

- APIs section
- SDKs section
- Guides section (reviewed each sub section except the ones listed below)

**Not Tested**
The following sections of this PR are not tested as these sections will be reviewed in a separate task. Only the base_path URL changes has currently been applied to these sections.

- Get Started
- Setup
- Guides
   - Authentication
   - Authorization
   - WS Trust
   - Applications
   - Identity Providers
   - References

**Not fixed by this PR**

Some broken hyperlinks in the following sections are not fixed by this PR:

- Jit provisioning links in the following:
    - Manage Passwords/Password Policies
    - User Management/Manage Accounts/Associate Accounts
    - User management/InvitationWorkflow
    - User management/Admin Creation Workflow
    - User management/Onboard Users overview
    - Outbound Provisioning Users/SCIM 2
    - Outbound Provisioning Users/Google Directory
    - Outbound Provisioning Users/Salesforce
    - Outbound Provisioning Users/Hubspot
- Links in the Configure Userstores for SCIM 1.1 section
- Links in the Configure AD Userstores for SCIM 1.1 section
- Links in the Sync User accounts

### CODE REVIEW

File: en/docs/developer-portal/applications/new-application.md [hunk 1]
```diff
@@ -4,10 +4,10 @@
 
 <div class="content">
```
Comment (nilminiwso2) [inline en/docs/developer-portal/applications/new-application.md:5]: @divyaamunugama - These files in the `developer-portal` older can be removed from the repo yeah?

File: en/docs/get-started/overview.md [hunk 2]
```diff
@@ -30,7 +30,7 @@ Basically, WSO2 Identity Server takes care of secure yet convenient sign-ins of
 
 ## Why WSO2 Identity Server?
 
-![IAM functionality](../../assets/img/guides/iam-functionality.png)
+![IAM functionality]({{base_path}}/assets/img/guides/iam-functionality.png)
```
Comment (nilminiwso2) [inline en/docs/get-started/overview.md:33]: @himeshsiriwardana Shall I remove this file from the repo?
I don't think it is used anywhere?

File: en/docs/developer-portal/applications/new-application.md (hunk 1 above)
Comment (divyaamunugama) [reply en/docs/developer-portal/applications/new-application.md:5]: yes, these can be removed

File: en/docs/get-started/overview.md (hunk 2 above)
Comment (himeshsiriwardana) [reply en/docs/get-started/overview.md:33]: +1

**Review by himeshsiriwardana — APPROVED** [review]

**Pull request merged**